pkg/server/handler/user.go

```diff
@@ -139,6 +139,32 @@ func (payload *userUpdatePayload) Validate() skyerr.Error {
 	return nil
 }
 
+// UserUpdateHandler update an UserInfo with the supplied information.
+//
+// UserUpdateHandler allow user to update his attributes with provided ID.
+// The attributes are following:
+//
+// * username (string, optional)
+// * email (string, optional)
+// * roles ([]string, optional)
+//
+// Missing of attributes will keep the attibute as it.
+// For `roles`, this API will create the missing roles in database and
+// assoicate it immediately with the user.
+//
+// Beside updating current user ID, this API also allow user with admin role
+// or API call using `master_key` to update other user attributes.
```

Comment:
“The authenticated user is allowed to update user attributes of the same user. Users with admin role (or requests authenticated with master key) is allowed to update user attributes of all users.”